src/Maestro/SubscriptionActorService/PullRequestActor.cs

```diff
@@ -1038,5 +1066,219 @@ private async Task<RepositoryBranch> GetRepositoryBranch(string repo, string bra
 
             return repoBranch;
         }
+
+        private static string GetNewBranchName(string targetBranch) => $"darc-{targetBranch}-{Guid.NewGuid()}";
+
+        #region Code flow subscriptions
+
+        /// <summary>
+        /// Alternative to ProcessPendingUpdatesAsync that is used in the code flow (VMR) scenario.
+        /// </summary>
+        private async Task<ActionResult<bool>> ProcessCodeFlowUpdatesAsync(
+            List<UpdateAssetsParameters> updates,
+            InProgressPullRequest? pr)
+        {
+            // TODO https://github.com/dotnet/arcade-services/issues/3378: Support batched PRs for code flow updates
+            if (updates.Count > 1)
+            {
+                updates = [..updates.DistinctBy(u => u.BuildId)];
+                if (updates.Count > 1)
+                {
+                    _logger.LogWarning("Code flow updates cannot be batched with other updates. Will process the last update only.");
+                }
+            }
+
+            var update = updates.Last();
+
+            CodeFlowStatus? codeFlowStatus = await _codeFlowState.TryGetStateAsync();
+
+            // The E2E order of things for is:
+            // 1. We send a request to PCS and wait for a branch to be created. We note down this in the codeflow status. We set a reminder.
+            // 2. When reminder kicks in, we check if the branch is created. If not, we repeat the reminder.
```

Comment:
We don't have a maximum repeat count. Do we consider introducing any? I think the branch-creating operation shouldn't fail, and eventually we should get a branch. But having a "stop mechanism" is always a good sign. For future consideration...

Reply:
Yes, a fair point. However this PR is unfortunately mostly about checking in the happy path where we parked the effort at.
We do have a follow up in the issue linked to this mentioning non-happy path scenario.